server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -295,52 +293,8 @@ private void validate(
-            boolean metadataFromRemoteStore = indexMetadataEntry.getValue().v1();
             if (indexMetadata.getSettings().getAsBoolean(SETTING_REMOTE_STORE_ENABLED, false)) {
-                if (metadataFromRemoteStore) {
-                    Set<String> graveyardIndexNames = new HashSet<>();
-                    Set<String> graveyardIndexUUID = new HashSet<>();
-                    Set<String> liveClusterIndexUUIDs = currentState.metadata()
-                        .indices()
-                        .values()
-                        .stream()
-                        .map(IndexMetadata::getIndexUUID)
-                        .collect(Collectors.toSet());
-
-                    currentState.metadata().indexGraveyard().getTombstones().forEach(tombstone -> {
-                        graveyardIndexNames.add(tombstone.getIndex().getName());
-                        graveyardIndexUUID.add(tombstone.getIndex().getUUID());
-                    });
-
-                    // Since updates to graveyard are synced to remote we should neven land in a situation where remote contain index
-                    // metadata for graveyard index.
-                    assert graveyardIndexNames.contains(indexName) == false : String.format(
-                        Locale.ROOT,
-                        "Index name [%s] exists in graveyard!",
-                        indexName
-                    );
-                    assert graveyardIndexUUID.contains(indexUUID) == false : String.format(
-                        Locale.ROOT,
-                        "Index UUID [%s] exists in graveyard!",
-                        indexUUID
-                    );
-
-                    // Any indices being restored from remote cluster state should not already be part of the cluster as this causes
-                    // conflict
-                    boolean sameNameIndexExists = currentState.metadata().hasIndex(indexName);
-                    boolean sameUUIDIndexExists = liveClusterIndexUUIDs.contains(indexUUID);
-                    if (sameNameIndexExists || sameUUIDIndexExists) {
-                        String finalErrorMsg = String.format(Locale.ROOT, errorMsg, indexName);
-                        logger.info(finalErrorMsg);
-                        throw new IllegalStateException(finalErrorMsg);
-                    }
-
-                    boolean isHidden = IndexMetadata.INDEX_HIDDEN_SETTING.get(indexMetadata.getSettings());
-                    createIndexService.validateIndexName(indexName, currentState);
-                    createIndexService.validateDotIndex(indexName, isHidden);
-                    shardLimitValidator.validateShardLimit(indexName, indexMetadata.getSettings(), currentState);
-                } else if (restoreAllShards && IndexMetadata.State.CLOSE.equals(indexMetadata.getState()) == false) {
+                if (restoreAllShards && IndexMetadata.State.CLOSE.equals(indexMetadata.getState()) == false) {
```

Comment:
lets move the respective validations for index restore and metadata restore above itself. This method is not needed anymore.